src/client/dfs/dfs.c

```diff
@@ -153,6 +155,47 @@ struct dfs_entry {
 	char		*value;
 };
 
+#define TX_OPEN(d, t, f)						\
+do {									\
+	if (d->use_dtx) {						\
+		rc = daos_tx_open(d->coh, &t, f, NULL);			\
+		if (rc) {						\
+			D_ERROR("daos_tx_open() failed (%d)\n", rc);	\
+			D_GOTO(out, rc);				\
```

Comment:
Not a big fan of macros with goto implicit labels that have to be defined in the function. That's bug prone IMO.
Could we use a static inline function instead and do error checking in the original function?